hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -897,28 +897,40 @@ public HoodieCleanMetadata clean(String cleanInstantTime, boolean scheduleInline
     if (!tableServicesEnabled(config)) {
       return null;
     }
-    final Timer.Context timerContext = metrics.getCleanCtx();
-    CleanerUtils.rollbackFailedWrites(config.getFailedWritesCleanPolicy(),
-        HoodieTimeline.CLEAN_ACTION, () -> rollbackFailedWrites(skipLocking));
-
-    HoodieTable table = createTable(config, hadoopConf);
-    if (config.allowMultipleCleans() || !table.getActiveTimeline().getCleanerTimeline().filterInflightsAndRequested().firstInstant().isPresent()) {
-      LOG.info("Cleaner started");
-      // proceed only if multiple clean schedules are enabled or if there are no pending cleans.
-      if (scheduleInline) {
-        scheduleTableServiceInternal(cleanInstantTime, Option.empty(), TableServiceType.CLEAN);
-        table.getMetaClient().reloadActiveTimeline();
+    HoodieCleanMetadata metadata;
+    HoodieInstant ownerInstant = null;
+    try {
+      if (!skipLocking) {
+        ownerInstant = new HoodieInstant(true, HoodieTimeline.CLEAN_ACTION, cleanInstantTime);
+        this.txnManager.beginTransaction(Option.of(ownerInstant), Option.empty());
+      }
+      final Timer.Context timerContext = metrics.getCleanCtx();
+      CleanerUtils.rollbackFailedWrites(config.getFailedWritesCleanPolicy(),
```

Comment:
even if we decide to go w/ locking, we should try and keep the locking to the mandatory critical section. for eg, we should lock only the rollbackFailedwrites and not the entire clean operation.

Reply:
You're right, I'll optimize it.